src/index/Index.Text.cpp

```diff
@@ -509,7 +509,7 @@ void Index::openTextFileHandle() {
 }
 
 // _____________________________________________________________________________
-string Index::wordIdToString(Id id) const { return _textVocab[id]; }
+const string& Index::wordIdToString(Id id) const { return _textVocab[id]; }
```

Comment:
Hmm, how does this not return a reference to a local variable? When the string is compressed, `AccessReturnType` is `std::string` and the result of `expandPrefix()`. Now this returns a `const string&` to it, but then that returned string immediately goes out of scope?!

Reply:
The `_textVocab` is not compressed and thus is able to return a valid reference.
I could also change this return value to be the proper AccessReturnType but I liked this interface more and as soon as we would  think about compressing the text vocabulary, the compiler will complain.